engine/apps/mobile_app/serializers.py

```diff
@@ -28,3 +29,12 @@ class Meta:
             "locale",
             "time_zone",
         )
+
+    def validate_going_oncall_notification_timing(self, going_oncall_notification_timing):
+        if going_oncall_notification_timing is not None:
+            if len(going_oncall_notification_timing) == 0:
+                raise serializers.ValidationError(detail="invalid timing options")
+            for option in going_oncall_notification_timing:
+                if option not in [opt[0] for opt in MobileAppUserSettings.NOTIFICATION_TIMING_CHOICES]:
```

Comment:
_nit_ I would move `[opt[0] for opt in MobileAppUserSettings.NOTIFICATION_TIMING_CHOICES]` outside of the `for` loop so that it doesn't have to be recalculated each loop